cdap-ui/app/cdap/main.js

```diff
@@ -36,17 +33,83 @@ import SplashScreen from './components/SplashScreen';
 import ConnectionExample from './components/ConnectionExample';
 import Experimental from './components/Experimental';
 import cookie from 'react-cookie';
-
-require('./styles/lib-styles.less');
-require('./styles/common.less');
+import {MyNamespaceApi} from './api/namespace';
+import Redirect from 'react-router/Redirect';
+import Router from 'react-router/BrowserRouter';
+import Match from 'react-router/Match';
+import Miss from 'react-router/Miss';
+import Store from './services/store/store';
 
 class CDAP extends Component {
   constructor(props) {
     super(props);
     this.props = props;
     this.version = '4.0.0';
+    this.namespace = Store.getState().selectedNamespace;
+    this.namespaceList = Store.getState().namespaceList;
+    this.pathname = props.pathname;
+    this.render = this.render.bind(this);
+    Store.subscribe(this.render);
+    this.state = {
+      selectedNamespace : ''
+    };
+  }
+
+  componentWillMount(){
+    var selectedNamespace;
+
+    //Polls for namespace data
+    MyNamespaceApi.pollList()
+      .subscribe((res) => {
+
+        if(res.length > 0){
+          for(var i = 0; i < res; i++){
+            if(res[i].description === 'Default Namespace'){
+              selectedNamespace = res[i].name;
+            }
+          }
+
+          if(!selectedNamespace){
+            selectedNamespace = res[0].name;
+          }
+
+          Store.dispatch({
+            type: 'UPDATE_NAMESPACES',
+            payload: {
+              namespaces : res
+            }
+          });
+
+          Store.dispatch({
+            type: 'SELECT_NAMESPACE',
```

Comment:
this will get triggered when the list of namespace gets updated. Is that intentional?